src/Entity/WorkspacesRouteLock.php

```
@@ -0,0 +1,119 @@
+ *     },
+ *   },
+ *   config_prefix = "workspaces_route_lock",
+ *   admin_permission = "administer site configuration",
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "label" = "label",
+ *     "uuid" = "uuid"
+ *   },
+ *   links = {
+ *     "canonical" = "/admin/config/workflow/workspaces/route_lock/{workspaces_route_lock}",
+ *     "add-form" = "/admin/config/workflow/workspaces/route_lock/add",
+ *     "edit-form" = "/admin/config/workflow/workspaces/route_lock/{workspaces_route_lock}/edit",
+ *     "delete-form" = "/admin/config/workflow/workspaces/route_lock/{workspaces_route_lock}/delete",
+ *     "collection" = "/admin/config/workflow/workspaces/route_lock"
+ *   },
+ *   config_export = {
+ *     "id",
+ *     "label",
+ *     "uuid",
+ *     "workspaces",
+ *   }
+ * )
+ */
+class WorkspacesRouteLock extends ConfigEntityBase implements WorkspacesRouteLockInterface {
+
+  /**
+   * The Workspaces route lock ID.
+   *
+   * @var string
+   */
+  protected $id;
+
+  /**
+   * The label.
+   *
+   * @var string
+   */
+  protected $label;
+
+  /**
+   * The list of allowed workspaces.
+   *
+   * @var string
+   */
+  protected $workspaces;
+
+  /**
+   * {@inheritDoc}
+   */
+  public function appliesToRoute(String $routeId) {
+    $routeMask = $this->label();
+    if ($routeId === $routeMask) {
+      // The route matches directly.
+      return TRUE;
+    }
+
+    // Escape the dots and turn '*' into '.*'.
+    $pattern = str_replace(['.', '*'], ["\.", ".*"], $routeMask);
+    return (bool) preg_match("/^$pattern$/", $routeId);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function workspaceIsAllowed(String $workspaceId) {
+    return in_array($workspaceId, $this->getWorkspaces());
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function getWorkspaceToSwitchTo() {
+    foreach ($this->getWorkspaces() as $workspaceId) {
+      $workspace = Workspace::load($workspaceId);
+      if ($workspace) {
+        return $workspace;
+      }
+    }
```

From the interface:
```   * @return \Drupal\workspaces\WorkspaceInterface```
But current implementation can return NULL.

Probably we should throw an exception here?